### ISSUE TITLE
allow USB hot-plugging

### ISSUE THREAD

**Issue by mariobuoninfante** [pure-data/pure-data#674]
I know this is quite a big one. 
Pretty sure someone already told me the issue is PortMidi (and I suppose PortAudio as well).
But still I think it's worth calling this out, especially cause nowadays swapping MIDI devices is not a corner case anymore.

**Comment by Spacechild1** [comment]
I agree. If this is still not possible with portmidi, we should patch portmidi :-)

**Comment by danomatika** [comment]
There is work on this upstream in port midi, but you'll have to wait for them to finish it.

**Comment by mariobuoninfante** [comment]
Nice, good to know they're working on that though :)

**Comment by Spacechild1** [comment]
This should now be possible with https://github.com/pure-data/pure-data/commit/816e5d70c37b1214bf4fb0322add37a4700e58d2, with the exception of macOS. But there is a dedicated issue for that: https://github.com/pure-data/pure-data/issues/2687#issuecomment-3080260852
